### ISSUE TITLE
grub fails to install

### ISSUE THREAD

**Issue by bfitzgit23** [manjaro/manjaro-tools#288]
hi there,

i just tried to build a manjaro ISO with the xfce version on the stable branch and keeps it looking for a module called overlay on the mkinitcpio portion and during the grub-probe portion. then it says failed to install correctly during the grub package installtion during the base system install. i've dealt with this and cannot simply ignore it per when i load my ISO during the build, i don't know how to fix this or why it's occuring. here is my error in a screenshot. https://i.imgur.com/OT5JHLA.png i don't know where the logs are, so i couldn't attach them so hopefully this is enough to get help. thanks.

**Comment by philmmanjaro** [comment]
You may check it with one of our [current build logs](https://osdn.net/projects/manjaro/storage/xfce/17.1.7-rc6/manjaro-xfce-17.1.7-rc6-testing-x86_64-build.log). It is totally normal that grub fails to execute the post-hook cmds and also intended during base installation:

> ( 99/225) installing grep                          [######################] 100%
(100/225) installing grub                          [######################] 100%
error: command failed to execute correctly
Optional dependencies for grub

As in a later step we create the actual boot menu for our ISO:

> ==> Prepare [/iso/boot]
 --> overlayfs mount: [/var/lib/manjaro-tools/buildiso/xfce/x86_64/bootfs]
  -> Copying initcpio ...
 --> Setting mirrorlist branch: testing
==> Starting build: 4.14.28-1-MANJARO
  -> Running build hook: [base]
  -> Running build hook: [udev]
  -> Running build hook: [miso_shutdown]
  -> Running build hook: [miso]
  -> Running build hook: [miso_loop_mnt]
  -> Running build hook: [miso_pxe_common]
  -> Running build hook: [miso_pxe_http]
  -> Running build hook: [miso_pxe_nbd]
  -> Running build hook: [miso_pxe_nfs]
  -> Running build hook: [miso_kms]
  -> Running build hook: [modconf]
  -> Running build hook: [block]
  -> Running build hook: [pcmcia]
  -> Running build hook: [filesystems]
  -> Running build hook: [keyboard]
  -> Running build hook: [keymap]
==> Generating module dependencies
==> Creating xz-compressed initcpio image: /boot/initramfs.img
bsdcpio: Failed to set default locale
==> Image generation successful
 --> overlayfs umount: [/var/lib/manjaro-tools/buildiso/xfce/x86_64/bootfs]
==> Done [/iso/boot]
==> Prepare [/iso/boot/grub]
  -> Building core.img ...
  -> Building bootx64.efi ...
  -> Creating fat image of 4M ...
 --> mount: [/var/lib/manjaro-tools/buildiso/xfce/mnt/efiboot]
  -> Building bootx64.efi ...
 --> umount: [/var/lib/manjaro-tools/buildiso/xfce/mnt/efiboot]
==> Done [/iso/boot/grub]
 --> Time prepare_images: 4.57 minutes
==> Start [Build ISO]

**Comment by philmmanjaro** [comment]
Only issue what I can think of is the removed **AUFS** support due this commit https://github.com/manjaro/manjaro-tools/commit/5ed1df3f39ab7b740e87bbabf834afdee9b9824e. Kernel-Series v3.16 doesn't support **overlayfs** yet. I've to see if we can backport that functionality.

**Comment by bfitzgit23** [comment]
Yeah I had this very error for months Phil I can’t seem to get round it. It’s weird I hope there’s a fix thanks.

**Comment by philmmanjaro** [comment]
Normally the manjaro-tools should fail:

```
phil@development /usr/lib/manjaro-tools $ buildiso -qv -k linux316
==> ERROR: The host and iso kernels must be version>=4.0!
```

Which version do you use?

**Comment by bfitzgit23** [comment]
I used the 4.14 kernel and I used virtualbox as one of my programs but that doesn’t explain why grub fails

**Comment by philmmanjaro** [comment]
Based on your log you used **linux316**, which is not supported by the current **manjaro-tools v0.15.5**. So I don't know how you managed that. Given by default we ship with **linux414** on our ISOs. What is your goal to create a custom ISO on your own?

**Comment by bfitzgit23** [comment]
Yes I’m trying to
Make a remix and I don’t wish to use 316 kernel

**Comment by philmmanjaro** [comment]
Somehow it ended up as the default kernel by a dependency cycle. Try to start the process with some like: `buildiso -k linux414 -p <your profile name>` and post the whole output of the terminal to gist or pastebin if you still fail. Somehow your profile setup is kinda wrong. You may check defaults with `buildiso -qv`

Example:

```
buildiso -p kde -qv
==> manjaro-tools
  -> version: 0.15.5
  -> config: ~/.config/manjaro-tools/manjaro-tools.conf
==> PROFILE:
  -> build_lists: community|default|manjaro|sonar
  -> build_list_iso: kde
  -> is_build_list: false
==> OPTIONS:
  -> arch: x86_64
  -> branch: unstable
  -> initsys: systemd
  -> kernel: linux414
==> ARGS:
  -> clean_first: true
  -> images_only: false
  -> iso_only: false
  -> persist: false
==> DIST SETTINGS:
  -> dist_name: Manjaro
  -> dist_release: 17.1.7-rc3
  -> dist_codename: Hakoila
==> ISO INFO:
  -> iso_label: MJRO1717
==> BUILD QUEUE:
 --> Profile: [kde]
  -> iso_file: manjaro-kde-17.1.7-rc3-unstable-x86_64.iso
  -> autologin: true
  -> nonfree_mhwd: true
  -> multilib: true
  -> extra: true
  -> netinstall: false
  -> chrootcfg: false
  -> geoip: true
  -> efi_boot_loader: grub
  -> hostname: manjaro
  -> username: manjaro
  -> password: manjaro
  -> login_shell: /bin/bash
  -> addgroups: video,power,storage,optical,network,lp,scanner,wheel,sys
  -> enable_systemd: avahi-daemon bluetooth cronie ModemManager NetworkManager org.cups.cupsd tlp tlp-sleep sddm
  -> enable_systemd_live: manjaro-live mhwd-live pacman-init mirrors-live
  -> disable_systemd: pacman-init
```

**Comment by bfitzgit23** [comment]
Yeah I found that I was using the wrong kernel and the wrong virtualbox host modules so I fixed it thanks
